src/modules/room/RoomSearchContainer.tsx

```diff
@@ -83,28 +106,35 @@ export function RoomSearchContainer() {
                   components={{
                     b: <strong />,
                   }}
-                  count={searchRoomQuery.data?.rooms.length || 0}
+                  count={searchRoomQuery.data?.pages[0]?.pagination?.total || 0}
                 />
-                {/* TODO: change to use pagination data after infinite scroll implemented */}
-                {/* <strong>Found {searchRoomQuery.data?.rooms.length}</strong>{' '}
-                rooms match your filters */}
               </Text>
-              {/* loading */}
+              {/* LOADING */}
               {searchRoomQuery.isLoading && <Spinner />}
-              {/* success with data */}
+
+              {/* DATA */}
               {!searchRoomQuery.isLoading &&
-                searchRoomQuery.data &&
-                searchRoomQuery.data?.rooms.length > 0 && (
+                searchRoomQuery.data?.pages &&
+                searchRoomQuery.data.pages.length > 0 && (
```

Comment:
is the `?.pages` come from the `useInfiniteQuery` hooks?

Reply:
yes @Fakhri1999 u can check it on https://tanstack.com/query/latest/docs/framework/react/guides/infinite-queries

Reply:
oohh okayy nicee